### ISSUE TITLE
This is so dope.

### ISSUE THREAD

**Issue by alexhawkins** [codefoundries/isomorphic-material-relay-starter-kit#68]
Well done. This is so dope. Dream stack. Watch the star count go nuts in a few months when people start to realize this is the future. How long before you have a working example with a real Cassandra DB? What would also be cool is if you set up the GraphQL server so it showed people how to use GraphQL with a microservice architecture. This is the only ding to GraphQL that I keep hearing and fact is, GraphQL works great with microservices from what I've read. " GraphQL can stand in front of a REST interface, for instance. Each piece of data can be fetched with a separate call to a separate microservice, executed in parallel and assembled into the structure the client wants."  I can prob figure this out in my free time if you guys are interested/need help. But this starter kit is 'bleeding from the edge of the event horizon' sick. You hit everything I would ever want at this juncture in time. Well done. Zero complaints. And that's unprecedented for me. I always have some complaint. Not here. Ok , throw in some tenserflow action and then I would be blown away. Pretty sure there is a node tensorflow lib https://github.com/node-tensorflow/node-tensorflow.

**Comment by thelordoftheboards** [comment]
Thank you very much for the compliments, highly appreciated.

In its current incarnation, the stack has quite a lot of custom code and hacks to go around the lack of certain capabilities in the underlying libraries. As time goes by, I am hoping that those things will get resolved, and the hacks can be removed. So in a sense the starter kit will becomes somewhat less necessary. But then again, if one is to start a project with Relay, Material-UI and Cassandra (and some of the other things you suggest) it will be an even better starting off point.

The examples currently in the kit do work with Cassandra, you can switch between in-memory and Cassandra implementation by running `config-da-memory` and `config-da-cassandra`.

The way the Node.js express-graphql is implemented, all the requests for data and mutations are implemented as javascript functions. For instance:

```javascript
    Ensayos: {
      type: EnsayosConnection.connectionType,
      args: { ...connectionArgs },
      resolve: ( obj, { ...args }, { rootValue: {user_id} } ) => DA_Ensayo_list_get( user_id ).then( ( arr_Ensayo ) => connectionFromArray( arr_Ensayo, args ) )
    },
```

In this case, the list of essays is retrieved by making a call to `DA_Ensayo_list_get`, which is simply a JavaScript function returning a promise. Luckily, both the `resolve` methods, and the `mutateAndGetPayload` methods used by mutations accept promises. So, they could just fire HTTP requests for micro services. If you like to add an example you are more than welcome to do so and I would be glad to merge a PR. I will only ask to keep it separate from the rest of the examples. If you notice all the examples are kind-of independent. My goal in doing this is that it is:

* Easy to remove all the examples and use the starter kit as a pure boiler plate
* Remove some of the example, and modify others and use them as starting point for a project.

As for tensor flow, I will be very honest, I am a bit behind the curve. I have only read about it and have not yet used it, even to run examples. I am a bit more familiar with Spark, and was planning to add some examples at some point. Tensor Flow is also a very promising toolkit.

**Comment by thelordoftheboards** [comment]
@alexhawkins have you had a change to see if the `resolve` and `mutateAndGetPayload` methods will work with micro services the way you intended? If so I will go ahead and close the issue.
Also, as a next step I will be implementing the [Facebook dataloader](https://github.com/facebook/dataloader) and it would make sense to put any data retrieval, from micro services, or not, behind the data loader to take advantage of caching.

**Comment by alexhawkins** [comment]
@thelordoftheboards I have not. I've been slammed with work but will be diving into this heavily in the very near term. Keep up the great work.

**Comment by alexhawkins** [comment]
@thelordoftheboards This is an interesting implementation of the RGR stack. Might give you some more ideas. Would be nice to get graphiql working with this build. https://github.com/ramsaylanier/WordPressExpress

**Comment by thelordoftheboards** [comment]
This is an interesting project, I saw it but have not had a time to study it yet.
For GraphiQL I am unable to get the server rendering - it literally fails in the include, so can't even make it conditional based on whether it is server or client rendering. I put an issue asking for help, I have no idea how to resolve this.

**Comment by thelordoftheboards** [comment]
For now I will keep the issue open and mark it as question until you get to test if the fix works or not.

**Comment by thelordoftheboards** [comment]
@alexhawkins I am going to close the issue at this point. If there is still a problem, please re-open.
I am also working on moving all the database access to dataloader, something I expect to finish by the end of next week, at which point I will assign label 1.0 and consider the starter kit good for production.
